### PR TITLE
use dict instead of if/else logic

### PR DESCRIPTION
**Pull request by r0fls**

### CODE REVIEW

File: flask/sessions.py [hunk 1]
```diff
@@ -84,21 +84,25 @@ class TaggedJSONSerializer(object):
     def dumps(self, value):
         return json.dumps(_tag(value), separators=(',', ':'))
 
+    LOADS_MAP = {
```
Comment (RonnyPfannschmidt) [inline flask/sessions.py]: Why lambda, just use the types

Comment (r0fls) [inline flask/sessions.py]: @RonnyPfannschmidt good point, thank you. Updated.

File: flask/sessions.py [hunk 2]
```diff
@@ -84,21 +84,25 @@ class TaggedJSONSerializer(object):
...
+        ' t': tuple,
+        ' u': uuid.UUID,
+        ' b': b64decode,
+        ' m': Markup,
+        ' d': parse_date
```
Comment (RonnyPfannschmidt) [inline flask/sessions.py]: little nitpick, please add a comma at the end, and  reindent the closing braces

that makes it more future friendly and a pleasure to read

File: flask/sessions.py [hunk 3]
```diff
@@ -84,21 +84,25 @@ class TaggedJSONSerializer(object):
     def dumps(self, value):
         return json.dumps(_tag(value), separators=(',', ':'))
 
+    LOADS_MAP = {
+        ' t': tuple,
+        ' u': uuid.UUID,
+        ' b': b64decode,
+        ' m': Markup,
+        ' d': parse_date,
+        }
```
Comment (r0fls) [inline flask/sessions.py]: @ThiefMaster kk, thank you for the specificity.

**Comment by davidism** [comment]
Haven't looked at it in a while, but make sure this is in line with #1452.

File: flask/sessions.py [hunk 4]
```diff
@@ -84,21 +84,25 @@ class TaggedJSONSerializer(object):
...
-            elif the_key == ' u':
-                return uuid.UUID(the_value)
-            elif the_key == ' b':
-                return b64decode(the_value)
-            elif the_key == ' m':
-                return Markup(the_value)
-            elif the_key == ' d':
-                return parse_date(the_value)
+            # Check the key for a corresponding function
+            return_function = self.LOADS_MAP.get(the_key, None)
```
Comment (tuukkamustonen) [inline flask/sessions.py]: iirc dict `get()` defaults to `None` so it can be omitted here (unless this is project's coding style preference)?

Comment (r0fls) [inline flask/sessions.py]: @tuukkamustonen updated

**Comment by r0fls** [comment]
This does appear to be made obsolete by https://github.com/pallets/flask/pull/1452, thanks @davidism. However that PR has been open for a year and a half, so this simple change might make sense while we're waiting for that to be approved. I'll leave that decision to the discretion of those with merge rights though.

**Comment by RonnyPfannschmidt** [comment]
@davidism at first glance #1452 looks like a over-generalization with quite a performance penalty :/

**Pull request merged**